exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MetricAdapter.java

```diff
@@ -225,6 +233,30 @@ private static AggregationTemporality mapToTemporality(
     return result;
   }
 
+  static Collection<DoubleHistogramDataPoint> toDoubleHistogramDataPoints(
+      Collection<DoubleHistogramPointData> points) {
+    List<DoubleHistogramDataPoint> result = new ArrayList<>(points.size());
+    for (DoubleHistogramPointData doubleHistogramPoint : points) {
+      DoubleHistogramDataPoint.Builder builder =
+          DoubleHistogramDataPoint.newBuilder()
+              .setStartTimeUnixNano(doubleHistogramPoint.getStartEpochNanos())
+              .setTimeUnixNano(doubleHistogramPoint.getEpochNanos())
+              .setCount(doubleHistogramPoint.getCount())
+              .setSum(doubleHistogramPoint.getSum())
+              .addAllBucketCounts(doubleHistogramPoint.getCounts());
+      List<Double> boundaries = doubleHistogramPoint.getBoundaries();
+      if (!boundaries.isEmpty()) {
+        builder.addAllExplicitBounds(boundaries);
+      }
```

Comment:
Counts as well. So we add boundaries and counts if boundaries not empty, otherwise the counts will be a list with one long equivalent with count.

Reply:
> counts will be a list with one long equivalent with count.

that looks like a valid OTLP histogram point to me according to https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L428.

boundaries and counts both being empty might be a problem to the downstream consumers since it's not in the contract.

Reply:
https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L418

Reply:
Comment is written with the mindset that there will be more options to define the boundaries like linear or exponential

Reply:
not sure if I follow. seems to me that the minimal size of `buckets` should be one according to https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L414

> // The number of elements in bucket_counts array must be by one greater than
  // the number of elements in explicit_bounds array.

and the service we built treats histogram data points without `buckets` invalid.